lib/soapforce/client.rb

```diff
@@ -19,6 +19,9 @@ def initialize(options = {})
       # to make SOAP calls in Professional/Group Edition organizations.
 
       client_id = options[:client_id] || Soapforce.configuration.client_id
+			
+			@headers = options[:headers] if options[:headers]
+				
       @headers = { 'tns:CallOptions' => { 'tns:client' => client_id } } if client_id
```

Comment:
This combination of `client_id` would not work with optional headers.  What if you merge the hashes? `@headers.merge(options[:headers]) if options[:headers].is_a?(Hash)`